### PR TITLE
Drop support for Python 2.7

### LINKED ISSUE #99: Drop support for Python 2.7
**Description of the desired feature**

With Unidata/MetPy#1163 on its way and other projects that depend on us not using 2.7 any more, I think we can aim to ditch 2.7 on our next release. We probably don't need many code changes. A few things that come to mind:

* Remove 2.7 builds from CIs
* Remove 2.7 dependencies and edit `setup.py`
* Remove import code for backports of some modules
* Edit the conda-forge recipe to build `noarch` since we no longer need different dependencies depending on Python version.

### PR DESCRIPTION
**Pull request by hugovk**
Python 2.7 reaches EOL on 2020-01-01 and many projects are dropping support for it.

Fixes #99.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [n/a] Add tests for new features or tests that would have caught the bug that you're fixing.
- [n/a] Add new public functions/methods/classes to `doc/api/index.rst`.
- [n/a] Write detailed docstrings for all functions/methods.
- [n/a] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

TODO from #99:

* [ ] Edit the conda-forge recipe to build noarch since we no longer need different dependencies depending on Python version.

I'm not sure when to do this.

### CODE REVIEW

**Comment by leouieda** [comment]
:wave: @hugovk thank you for the contribution!

> Edit the conda-forge recipe to build noarch since we no longer need different dependencies depending on Python version.

This would be done after the release only, so no need to worry about that now.

File: .travis.yml [hunk 1]
```diff
@@ -3,9 +3,6 @@
 # We use miniconda for Python so don't need any Python specific tools
 language: generic
 
-# Use the container builds so we don't need sudo priviledges
-sudo: false
-
```
Comment (leouieda) [inline .travis.yml]: Nice catch!

**Comment by leouieda** [comment]
Alright! Nice to see all checks green :+1: I'll merge this in shortly. Just need to verify that we can drop 2.7 on master right now.

**Comment by leouieda** [comment]
@hugovk thank you for your contribution to Pooch! :confetti_ball: :clinking_glasses: We're always looking for feedback on the contributing process so any inputs would be very welcome. Feel free to add yourself to the `AUTHORS.md` file (if you wish).

**Comment by hugovk** [comment]
Thanks, added!

The contributing process was smooth, there was a clear checklist in the PR template, and plenty of extra info in the contributing guidelines. I only skim-read those as I was removing/refactoring, but looks like there's good info there for creating new features.

**Comment by leouieda** [comment]
Thanks for the feedback! And hope to have you involved in the future :1st_place_medal:

**Pull request merged**